### DISCUSSION TITLE
Ollama local models context token length always displays 4096 even when LLMs support more

### DISCUSSION THREAD

**Discussion by mysterium-coniunctionis** [mckaywrigley/chatbot-ui#1297] [Help]
For some reason, local Ollama models that support contexts larger than 4096 tokens do not seem to adjust to the greater token length on Chatbot-UI. Has this been addressed anywhere? If yes, will you point me in the right direction to fix this on my localhost instance?

**Comment by mckaywrigley** [comment]
Do you know if the Ollama API gives us model context length? If so I can fix this in like 5min.

**Reply by mysterium-coniunctionis** [reply]
There are a few that vary as far as I can tell at various levels - as low as 2K and as high as 128K. I found a couple that were easy to search for and I'll add more to this thread as I have a bit of time to dig through the docs @ https://ollama.ai/library.

_I'm not sure if this is what you mean, but here are a few:_

**yarn-mistral & yarn-llama2 both come in 64K & 128K versions**

<img width="532" alt="Screenshot 2024-01-28 at 6 24 32 PM" src="https://github.com/mckaywrigley/chatbot-ui/assets/3911621/8ebfee70-4f14-4925-b30a-86aa9ae4868d">

**vicuna comes in 2K-16K versions**

<img width="524" alt="Screenshot 2024-01-28 at 6 24 52 PM" src="https://github.com/mckaywrigley/chatbot-ui/assets/3911621/ba4b867c-c003-45f9-9369-3599b56b940a">

**everythinglm comes a 16K version**

<img width="475" alt="Screenshot 2024-01-28 at 6 25 00 PM" src="https://github.com/mckaywrigley/chatbot-ui/assets/3911621/1b4b2eab-505b-4c76-86f6-cc5d2126a24d">

**alfred comes in an 8K version**

<img width="902" alt="Screenshot 2024-01-28 at 6 26 18 PM" src="https://github.com/mckaywrigley/chatbot-ui/assets/3911621/d51b2719-7576-4ba9-82a8-17ebfe40f77d">
